cmd/aries-agentd/startcmd/start_test.go

```diff
@@ -0,0 +1,344 @@
+	require.Len(t, requiredFlagAnnotation, 1)
+	require.Equal(t, requiredFlagAnnotation[0], "true")
+
+	return
+}
+
+func TestStartAriesDRequests(t *testing.T) {
+	// TODO - remove this path manipulation after implementing #175 and #148
+	path, cleanup := generateTempDir(t)
+	defer cleanup()
+
+	testHostURL := randomURL()
+	testInboundHostURL := randomURL()
+	go startAgent(&RealServer{}, testHostURL, testInboundHostURL, path)
+	// give some time for server to start
+	if err := listenFor(testHostURL, time.Second); err != nil {
+		t.Fatal(err)
+	}
+	if err := listenFor(testInboundHostURL, time.Second); err != nil {
+		t.Fatal(err)
+	}
+	validateRequests(t, testHostURL, testInboundHostURL)
+}
+
+func listenFor(host string, d time.Duration) error {
+	timeout := time.After(d)
+	for {
+		select {
+		case <-timeout:
+			return errors.New("timeout: server is not available")
+		default:
+			conn, err := net.Dial("tcp", host)
+			if err != nil {
+				continue
+			}
+			if err := conn.Close(); err != nil {
+				return err
+			}
+			return nil
+		}
+	}
+}
+
+// TODO: this method should be refactored (e.g., too many lines).
+//nolint:funlen
+func validateRequests(t *testing.T, testHostURL string, testInboundHostURL string) {
+	newreq := func(method, url string, body io.Reader, contentType string) *http.Request {
+		r, err := http.NewRequest(method, url, body)
+		if contentType != "" {
+			r.Header.Add("Content-Type", contentType)
+		}
+		if err != nil {
+			t.Fatal(err)
+		}
+		return r
+	}
+
+	tests := []struct {
+		name               string
+		r                  *http.Request
+		expectedStatus     int
+		expectResponseData bool
+	}{
+		// controller API test
+		{
+			name:               "1: testing get",
+			r:                  newreq("GET", fmt.Sprintf("http://%s/connections/create-invitation", testHostURL), nil, ""),
+			expectedStatus:     http.StatusOK,
+			expectResponseData: true,
+		},
+
+		// DIDComm inbound API test
+		{
+			name: "200: testing didcomm inbound",
+			r: newreq("POST",
+				fmt.Sprintf("http://%s", testInboundHostURL),
+				strings.NewReader(`
+							{
+								"@id": "5678876542345",
+								"@type": "did:sov:BzCbsNYhMrjHiqZDTUASHg;spec/didexchange/1.0/invitation"
+							}`),
+				"application/didcomm-envelope-enc"),
+			expectedStatus:     http.StatusInternalServerError,
+			expectResponseData: false,
+		},
+	}
+
+	for _, tt := range tests {
+
+		resp, err := http.DefaultClient.Do(tt.r)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		defer func() {
+			e := resp.Body.Close()
+			if e != nil {
+				panic(err)
+			}
+		}()
+
+		require.Equal(t, tt.expectedStatus, resp.StatusCode)
+		if tt.expectResponseData {
+			require.NotEmpty(t, resp.Body)
+			response, err := ioutil.ReadAll(resp.Body)
+			if err != nil {
+				t.Fatal(err)
+			}
+			require.NotEmpty(t, response)
+			require.True(t, isJSON(response))
+		}
+	}
+
+}
+
+// isJSON checks if response is json
+func isJSON(res []byte) bool {
```

Comment:
use empty struct
```
func isJSON(res []byte) bool {
	var js struct{}
	return json.Unmarshal(res, &js) == nil
}
```

Reply:
This was code that I didn't originally write - I just moved it from another file to here. It was an easy change to make though, so I did it.

Reply:
thats function supposed to find if given bytes is in JSON format, not sure empty struct can help here. `map[string]interface{}` makes sure if given bytes are in JSON format @soluchok

Reply:
For now I'll revert back to the original way. Since I didn't originally write this code I think it's probably safer to leave as is.